### PR TITLE
fileservice: remove unused codes

### PR DESCRIPTION
**Pull request by reusee**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #14908 

## What this PR does / why we need it:
clean up unused codes.

### CODE REVIEW

**Comment by matrix-meow** [comment]
@reusee Thanks for your contributions!

**Here are review comments for file cmd/mo-service/debug.go:**
### Pull Request Review:

#### Title and Body:
The title and body of the pull request clearly state the purpose of the changes, which is to remove unused code. It is important to clean up unused code to improve code maintainability and reduce potential confusion for developers. However, it is essential to ensure that the removal of code does not introduce any issues or affect the functionality of the application.

#### Changes in cmd/mo-service/debug.go:
1. **Unused Imports Removed:**
   - The import statement for `github.com/matrixorigin/matrixone/pkg/fileservice` has been removed. This is a good practice to eliminate unnecessary dependencies and improve code readability.

2. **Unused Flags and Functions Removed:**
   - Several flags related to profiling and a function `startFileServiceProfile` have been removed. These were likely unused or redundant, and removing them simplifies the codebase.

3. **HTTP Handler Removed:**
   - The registration of the `/debug/fs/` endpoint for `fileservice.FSProfileHandler` has been removed. Since the `fileservice` package is no longer imported, this handler is no longer needed.

#### Suggestions for Improvement:
1. **Code Review and Testing:**
   - It is crucial to conduct a thorough code review to ensure that no essential functionality is inadvertently removed. Additionally, testing should be performed to verify that the application behaves as expected after the removal of the unused code.

2. **Documentation Update:**
   - If any public APIs or configurations were removed, it is advisable to update the documentation or inform relevant stakeholders about these changes to prevent any unexpected issues.

3. **Optimization Opportunity:**
   - While the removal of unused code is beneficial, consider conducting a broader analysis of the codebase to identify other areas where optimization or refactoring could further enhance the application's performance and maintainability.

4. **Security Consideration:**
   - Although not directly related to this pull request, it is essential to regularly review the codebase for security vulnerabilities and follow secure coding practices to mitigate potential risks.

By addressing the suggestions provided and ensuring thorough testing, the codebase can be improved in terms of readability, maintainability, and performance.

**Here are review comments for file cmd/mo-service/main.go:**
### Pull Request Review:

#### Title:
The title of the pull request is clear and concise, indicating the intention to remove unused code from the fileservice.

#### Body:
The body of the pull request provides some context about the type of PR and the issue it addresses. It mentions that the PR is an improvement and is related to cleaning up unused code for issue #14908.

#### Changes:
The changes in the pull request involve removing some code related to file service profiling in the `main.go` file of the `mo-service` command. Specifically, the code block responsible for starting and stopping the file service profile has been removed.

### Feedback and Suggestions:

1. **Unused Code Removal:**
   The removal of unused code is generally a good practice as it helps in reducing complexity and improves code maintainability. In this case, the code related to file service profiling has been removed, which is a positive step.

2. **Optimization:**
   Since the code related to file service profiling has been removed, it would be beneficial to ensure that there are no other dependencies or references to this code elsewhere in the project. Performing a thorough search to confirm complete removal of all related code would be advisable.

3. **Documentation Update:**
   It is recommended to update any relevant documentation or comments that may have referenced the removed code. This ensures that the documentation remains accurate and up to date with the codebase.

4. **Code Review:**
   While removing unused code is beneficial, it is essential to conduct a comprehensive code review to verify that the removal does not inadvertently impact any other functionalities or introduce new issues. Thorough testing after the code removal is also crucial to ensure the stability of the application.

5. **Security Consideration:**
   Although not explicitly mentioned in the provided diff, it is crucial to consider any security implications of removing code, especially if it was related to profiling or performance monitoring. Ensure that the removal does not compromise any security measures or monitoring capabilities in the application.

6. **Version Control:**
   It is good practice to ensure that the removal of code is reflected in version control commit messages for clarity and traceability purposes.

### Overall, the pull request is on the right track by removing unused code. However, it is important to conduct a thorough review, update documentation, and consider any security implications before merging the changes.

**Here are review comments for file go.mod:**
### Pull Request Review:

#### Title:
The title of the pull request is clear and concise, indicating the intention to remove unused code from the fileservice.

#### Body:
The body of the pull request provides additional context by specifying the type of PR, the related issue, and the purpose of the changes, which is to clean up unused code.

#### Changes in go.mod:
1. **Removal of `github.com/google/pprof` dependency:**
   - The pull request removes the `github.com/google/pprof` dependency from the `go.mod` file as it is no longer being used.
   - **Issue:** While removing unused dependencies is generally a good practice, it's crucial to ensure that the dependency is truly not needed. Removing a dependency like `pprof` could potentially impact performance monitoring and profiling capabilities in the application.
   - **Suggestion:** Before removing the `pprof` dependency, it's recommended to confirm with the team whether it is indeed unused and if there are no plans to utilize it in the future. If there is uncertainty, it might be better to keep the dependency for now.

2. **Optimization Opportunity:**
   - Instead of just removing the dependency, consider adding a comment explaining why it was removed. This can help future developers understand the reasoning behind the change.

#### Overall Suggestions:
- **Code Review Process:** It's important to have a thorough code review process in place to ensure that dependencies are removed intentionally and not accidentally.
- **Documentation:** Consider updating the project documentation or README file to reflect any significant changes like dependency removals for better transparency.

#### Conclusion:
The pull request effectively removes the unused `github.com/google/pprof` dependency. However, it's essential to double-check the necessity of the dependency before removal and provide clear explanations for such changes to maintain code clarity and transparency.

**Here are review comments for file go.sum:**
### Pull Request Review:

#### Title:
The title of the pull request is clear and concise, indicating the intention to remove unused code from the fileservice.

#### Body:
The body of the pull request provides some context about the type of PR and the issue it addresses, which is good. It states that the purpose of the PR is to clean up unused code.

#### Changes:
The changes in the `go.sum` file show the removal of the `github.com/petermattis/goid` module, which suggests that this code is no longer being used and has been removed as part of the cleanup process.

#### Feedback:
1. **Unused Code Removal:**
   - The removal of unused code is a good practice as it helps in reducing code clutter and potential confusion for developers. It also improves the maintainability of the codebase.
   - It is important to ensure that the code being removed is genuinely unused and not required for any functionality. Double-check to confirm that removing `github.com/petermattis/goid` does not impact any existing functionality.

2. **Security Implications:**
   - While removing unused code is generally safe, it is crucial to conduct a thorough review to ensure that the removal does not introduce any security vulnerabilities.
   - Make sure that the removal of `github.com/petermattis/goid` does not inadvertently impact any security-related functionality or introduce any new vulnerabilities.

3. **Optimization:**
   - Consider conducting a broader review of the codebase to identify and remove any other unused code or dependencies. This can further streamline the codebase and improve its overall quality.
   - Utilize tools like static code analyzers to automatically detect and highlight unused code segments for removal.

4. **Documentation:**
   - It would be beneficial to update any relevant documentation or comments to reflect the removal of the unused code. This helps in maintaining consistency and clarity for future developers.

#### Overall:
The pull request to remove unused code is a positive step towards codebase cleanliness and maintainability. Ensure that the removal is thoroughly reviewed for any potential impacts, especially in terms of security. Consider additional optimizations and documentation updates to enhance the overall quality of the codebase.

**Here are review comments for file pkg/fileservice/aliyun_sdk.go:**
### Pull Request Review:

#### Title:
The title of the pull request is clear and concise, indicating the purpose of the changes made.

#### Body:
The body of the pull request provides some context about the type of PR and the issue it addresses. It mentions that the PR is an improvement to clean up unused code.

#### Changes:
The changes in the pull request involve removing some unused code snippets from the `aliyun_sdk.go` file in the `fileservice` package. The removed code includes deferred functions that were used for performance profiling but are no longer necessary.

#### Feedback and Suggestions:
1. **Unused Code Removal:**
   - The removal of unused code is a good practice as it helps in keeping the codebase clean and maintainable.
   - The deferred functions related to performance profiling (e.g., `FSProfileHandler.AddSample`) have been removed. This is a positive change as it eliminates unnecessary code execution.

2. **Optimization:**
   - While removing unused code is beneficial, it's important to ensure that the removal does not impact any essential functionality or logging.
   - It would be good to double-check if any other parts of the codebase were relying on the removed deferred functions for any critical operations.

3. **Code Review Best Practices:**
   - It's recommended to have a thorough code review process in place to catch and address such unused code issues before they get merged into the codebase.
   - Consider adding comments or annotations in the code to explain the reason for removing certain sections, especially if they were initially added for specific purposes.

4. **Testing:**
   - After removing the unused code, it's essential to run tests to ensure that the functionality of the `AliyunSDK` class or related components has not been affected.
   - Automated tests can help in detecting regressions caused by code changes.

5. **Documentation:**
   - Update any relevant documentation or comments to reflect the changes made in the codebase. This helps in maintaining consistency and clarity for future developers.

6. **Security Considerations:**
   - While the changes in this pull request seem focused on code cleanup, it's crucial to always be vigilant for any security implications when modifying code.
   - Ensure that the removal of code does not introduce any vulnerabilities or weaken the security posture of the application.

In conclusion, the pull request to remove unused code is a positive step towards codebase maintenance. By following the suggestions provided above, the code quality can be further improved, and the risk of introducing issues can be minimized.

**Here are review comments for file pkg/fileservice/aws_sdk_v1.go:**
### Pull Request Review:

#### Title:
The title of the pull request is clear and concise, indicating the intention to remove unused code from the fileservice.

#### Body:
The body of the pull request provides information about the type of PR, the related issue, and the purpose of the changes, which is to clean up unused code.

#### Changes:
The changes in the pull request involve removing code related to time tracking and performance profiling from various functions in the `AwsSDKv1` struct in the `aws_sdk_v1.go` file of the `fileservice` package.

#### Feedback:

1. **Unused Code Removal:**
   - The removal of the time tracking and performance profiling code seems appropriate if it is truly unused. However, it's essential to ensure that these functionalities are not needed elsewhere in the codebase before removing them.
   - **Suggestion:** Before deleting the code, double-check if there are any dependencies or future plans that might require this functionality. If not, the removal is justified.

2. **Code Duplication:**
   - The code snippets being removed from each function (`listObjects`, `headObject`, `putObject`, `getObject`, `deleteObject`, `deleteObjects`) are similar. This indicates code duplication.
   - **Suggestion:** Consider refactoring the code to have a common function for time tracking and performance profiling to avoid duplication and make future maintenance easier.

3. **Logging or Monitoring:**
   - Removing the time tracking and performance profiling code may impact monitoring or logging capabilities in the application.
   - **Suggestion:** If monitoring or logging is necessary, consider integrating a more centralized and consistent approach to handle such requirements across the codebase.

4. **Security Concerns:**
   - While the removal of the code itself does not introduce security vulnerabilities, it's crucial to ensure that no critical monitoring or profiling mechanisms are being eliminated inadvertently.
   - **Suggestion:** Conduct thorough testing to verify that the removal of this code does not impact the security or stability of the application.

5. **Documentation:**
   - It would be beneficial to update the relevant documentation or comments to reflect the changes made in this pull request.
   - **Suggestion:** Add a brief comment in the code explaining the reason for removing the time tracking and performance profiling code for future reference.

#### Overall, the removal of unused code is a good practice for code maintenance. However, it's important to ensure that the removed code is genuinely unnecessary and that the changes do not have unintended consequences on the application's functionality or monitoring capabilities. Additionally, consider refactoring to avoid code duplication and maintain consistency in the codebase.

**Here are review comments for file pkg/fileservice/aws_sdk_v2.go:**
### Pull Request Review:

#### Title:
The title of the pull request is clear and concise, indicating the purpose of the changes made.

#### Body:
The body of the pull request provides some context about the type of PR and the issue it fixes. It also mentions that the PR is for cleaning up unused code.

#### Changes:
The changes in the pull request involve removing some code snippets related to performance profiling in the `AwsSDKv2` file of the `fileservice` package.

#### Feedback:

1. **Unused Code Removal:**
   - The changes made in this pull request involve removing code related to performance profiling (`FSProfileHandler.AddSample`) which is no longer being used.
   - **Issue:** While removing unused code is generally a good practice, it's important to ensure that the removed code is truly unnecessary and does not have any impact on the functionality or performance monitoring of the application.
   - **Suggestion:** Before removing such code, it's advisable to double-check if there are any dependencies or reasons for keeping it. If not, the removal can proceed.

2. **Performance Monitoring Impact:**
   - The removed code snippets were responsible for tracking the time taken by certain operations using `time.Since(t0)` and updating performance counters.
   - **Issue:** Removing this code may impact performance monitoring and profiling capabilities, making it harder to track and optimize the performance of the application.
   - **Suggestion:** If performance monitoring is still required, consider replacing the removed code with a more efficient or up-to-date method for tracking performance metrics.

3. **Code Optimization:**
   - The repetitive pattern of starting and ending tasks and measuring time (`time.Now()`, `defer`, `time.Since(t0)`) has been removed from multiple functions.
   - **Optimization:** Instead of removing this code entirely, consider refactoring it into a reusable function or method to avoid code duplication and maintain a consistent approach to performance monitoring.

4. **Documentation Update:**
   - It would be beneficial to update the relevant documentation or comments to reflect the changes made in the code.
   - **Suggestion:** Add a comment or update the documentation to explain why the performance profiling code was removed and how performance monitoring will be handled moving forward.

5. **Testing:**
   - Ensure that after the removal of the code, the functionality of the `AwsSDKv2` methods is still intact and that no unintended side effects have been introduced.
   - **Recommendation:** Run relevant tests to verify that the removal of the performance profiling code does not break any existing functionality.

In conclusion, while cleaning up unused code is a good practice, it's essential to ensure that the removal does not have unintended consequences on performance monitoring or other aspects of the application. Consider the suggestions provided to optimize the changes and maintain the codebase's integrity and functionality.

**Here are review comments for file pkg/fileservice/disk_cache.go:**
### Pull Request Review:

#### Problem 1: Removing `time` package without updating related code
- The pull request removes the `time` package import and related code (`time.Now()` and `time.Since()`) from `disk_cache.go` file. However, it fails to update the `cacheHit` function that uses `time.Duration` as a parameter. This will cause a compilation error as `time.Duration` is no longer available due to the removal of the `time` package.
  
#### Solution 1:
- Update the `cacheHit` function in `disk_cache.go` to remove the reference to `time.Duration` since the `time` package has been removed. You can simply remove the parameter `duration time.Duration` and the call to `FSProfileHandler.AddSample(duration)`.

#### Problem 2: Incomplete removal of `cacheHit` function
- Although the `cacheHit` function is no longer being used after removing the `time` package, the function itself is not removed from the codebase. This can lead to confusion for future developers and unnecessary clutter in the code.
  
#### Solution 2:
- Consider removing the `cacheHit` function entirely from the `disk_cache.go` file since it is no longer being utilized. This will help in keeping the codebase clean and maintainable.

#### Problem 3: Lack of descriptive commit message
- The commit message "fileservice: remove unused codes" is too generic and does not provide detailed information about the changes made in the pull request. A more descriptive message would help other developers understand the purpose of the changes without having to inspect the code diff.

#### Solution 3:
- Provide a more informative commit message that explains the specific unused codes being removed and the reason behind their removal. This will improve the clarity and transparency of the pull request for reviewers and future reference.

#### Optimization:
- Consider performing a thorough code review to identify and remove any other unused code or redundant functions in the `disk_cache.go` file to further optimize the codebase and improve its maintainability.

By addressing the mentioned problems and optimizing the changes, the pull request can be enhanced to ensure a cleaner and more efficient codebase.

**Here are review comments for file pkg/fileservice/local_fs.go:**
### Pull Request Review:

#### Title and Body:
The title and body of the pull request clearly state the purpose of the changes, which is to remove unused code. It references the related issue and provides a brief explanation of the cleanup. However, it would be beneficial to include more specific details about the unused code being removed to provide better context for reviewers.

#### Changes in local_fs.go:
1. **Unused Code Removal:**
   - The pull request removes code related to timing measurements using `time.Now()` and `defer` statements in multiple functions (`write`, `read`, `List`, `StatFile`, `Delete`, `mutate`) within the `LocalFS` struct. These timing measurements were used to profile the file system operations but are now commented out or removed entirely.
   - **Issue:** The removal of these timing measurements might be necessary if they are no longer used for profiling or debugging purposes. However, if these measurements are still required for performance monitoring or debugging in the future, their removal could be a loss of valuable information.
   - **Suggestion:** Consider revisiting the decision to remove these timing measurements. If they are no longer needed, the removal is justified. Otherwise, if they serve a purpose, they should be retained or refactored for better clarity.

2. **Optimization Opportunity:**
   - In each function, the `defer` statement for adding samples to `FSProfileHandler` after measuring time can be optimized by moving the time measurement and defer statement into a separate function to reduce code duplication and improve maintainability.
   - **Suggestion:** Create a helper function to handle the time measurement and defer logic to avoid repeating the same code in multiple functions. This will enhance code readability and maintainability.

3. **Consistency:**
   - The removal of the timing measurements is consistent across all functions, which ensures uniformity in the codebase.
   - **Suggestion:** Ensure that any similar unused code or redundant patterns are reviewed and addressed consistently throughout the codebase for better code hygiene.

#### Overall Suggestions:
- Provide more detailed information in the pull request description about why the timing measurements were removed to aid in understanding the decision-making process.
- Consider documenting the reason for removing the unused code in code comments or in the pull request description for future reference.
- Ensure that the changes align with the project's coding standards and guidelines.

By addressing the mentioned points and considering the suggestions provided, the pull request can be improved in terms of clarity, maintainability, and adherence to best practices.

**Here are review comments for file pkg/fileservice/mem_cache.go:**
### Pull Request Review:

#### Problem 1: Incomplete Explanation in Pull Request Body
The pull request title and body mention that the purpose of the PR is to "clean up unused codes." However, the body lacks specific details about the unused code being removed. Providing more context in the body of the pull request would help reviewers understand the changes better.

#### Suggestion:
- Update the pull request body to include a brief explanation of the specific unused code that is being removed. This will provide more clarity to reviewers and future developers.

#### Problem 2: Removal of `cacheHit` Function
The `cacheHit` function in the `MemCache` struct has been removed in this pull request. This function was previously used to add a sample duration to `FSProfileHandler`. Removing this function may impact the profiling or monitoring capabilities of the codebase.

#### Suggestion:
- Instead of completely removing the `cacheHit` function, consider refactoring it to be more useful or repurpose it for any potential future profiling needs. If the function is truly unnecessary, ensure that there are no dependencies on it before removing it.

#### Problem 3: Lack of Comments or Documentation
The changes made in the `mem_cache.go` file lack comments or documentation to explain the purpose of the code being modified or removed. This can make it difficult for other developers to understand the intent behind the changes.

#### Suggestion:
- Add comments or documentation to explain the rationale behind the removal of the unused code. Documenting the code will help future developers understand the context and reasons for the changes made.

#### Optimization:
- Consider adding a comment above the removed `cacheHit` function explaining why it was deemed unnecessary and what alternative approach, if any, is being used.
- Ensure that the commit message clearly states the reason for removing the unused code to provide a comprehensive understanding for anyone reviewing the commit history.

By addressing the mentioned problems and suggestions, the pull request can be improved in terms of clarity, maintainability, and ensuring that the removal of code does not inadvertently impact the functionality or monitoring aspects of the application.

**Here are review comments for file pkg/fileservice/minio_sdk.go:**
### Pull Request Review:

#### Title:
The title of the pull request is clear and concise, indicating the intention to remove unused code from the fileservice.

#### Body:
The body of the pull request provides additional context by mentioning the type of PR (Improvement), the specific issue it addresses (issue #14908), and the purpose of the PR, which is to clean up unused code.

#### Changes:
The changes in the `minio_sdk.go` file involve removing code related to performance profiling that is no longer being used. The removal of the timing and profiling code snippets from various functions like `listObjects`, `statObject`, `putObject`, `getObject`, `deleteObject`, and `deleteObjects` is aimed at eliminating unnecessary overhead.

#### Feedback and Suggestions:
1. **Code Cleanup**: The removal of unused code is a good practice to maintain a clean and efficient codebase. However, it's important to ensure that the removed code is truly unnecessary and does not have any hidden dependencies or implications.

2. **Optimization Opportunity**: While removing the timing and profiling code snippets can reduce clutter, consider if there are any other optimizations or improvements that can be made in these functions to enhance performance or readability.

3. **Version Control**: It's crucial to have a version control system in place to track changes like these. This ensures that if the removed code is needed in the future or if any issues arise due to its removal, it can be easily reverted.

4. **Documentation**: Update any relevant documentation or comments to reflect the changes made in the code. This helps in maintaining clarity for other developers who might work on this code in the future.

5. **Testing**: After removing the code, it's advisable to run tests to ensure that the functionality of the affected functions has not been impacted by the removal of the profiling code.

6. **Security Concerns**: While the changes in this PR seem focused on code cleanup, it's always good to review the entire codebase for any potential security vulnerabilities or sensitive data handling issues.

By addressing the above points, the pull request can not only achieve the goal of removing unused code but also ensure that the codebase remains optimized and maintainable.

**Here are review comments for file pkg/fileservice/profile.go:**
### Pull Request Review:

#### Title:
The title of the pull request is clear and concise, indicating the intention to remove unused code from the fileservice.

#### Body:
The body of the pull request provides additional context by specifying the type of PR (Improvement), the related issue (#14908), and the purpose of the PR, which is to clean up unused code.

#### Changes:
The changes in the pull request involve the removal of a significant portion of code from the `profile.go` file in the `fileservice` package. The removed code includes licensing information, package imports, type definitions, functions, and related logic that seems to be related to profiling functionality.

#### Feedback and Suggestions for Improvement:
1. **Unused Code Removal:**
   - The removal of unused code is a good practice to keep the codebase clean and maintainable. However, it's essential to ensure that the removed code is genuinely unused and not required for any functionality.
   - Before removing code, it's advisable to perform a thorough analysis to confirm that the code is not being used anywhere in the application.

2. **License Information:**
   - While removing licensing information from the code may not impact the functionality directly, it's crucial to ensure that the appropriate licensing information is retained in the project repository or files as required by the project's licensing terms.

3. **ProfileHandler and Profiler Logic:**
   - The `ProfileHandler` and `profiler` logic seems to be related to profiling functionality. If profiling is a required feature in the application, it's important to verify if this functionality is truly unused or if it is needed for performance monitoring or debugging purposes.
   - If profiling is indeed unnecessary, the removal of this logic is appropriate. However, if profiling is needed, consider refactoring or optimizing the existing code instead of removing it entirely.

4. **Code Optimization:**
   - While removing unused code is beneficial, it's also important to optimize the existing codebase for better performance and maintainability.
   - Consider refactoring or restructuring the code to improve readability, reduce complexity, and enhance overall code quality.

5. **Documentation:**
   - It's recommended to update any relevant documentation or comments to reflect the changes made in the codebase. This helps in maintaining consistency and clarity for future developers.

6. **Security Concerns:**
   - While the changes in this pull request focus on code cleanup, it's crucial to ensure that no security vulnerabilities are introduced during the process.
   - Conduct a security review to identify and address any potential security issues, especially when modifying core functionalities like profiling.

In conclusion, while the removal of unused code is a positive step towards codebase maintenance, it's essential to verify the necessity of the removed code and consider potential impacts on functionality and performance. Additionally, optimizing the existing code and ensuring documentation updates can further enhance the quality of the codebase.

**Here are review comments for file pkg/fileservice/profile_info.go:**
### Pull Request Review:

#### Title:
The title of the pull request is clear and concise, indicating the intention to remove unused code from the fileservice.

#### Body:
The body of the pull request provides information about the type of PR, the related issue, and the purpose of the changes, which is to clean up unused code. It would be beneficial to include more details about the specific unused code being removed for better context.

#### Changes in profile_info.go:
1. **License Information:**
   - The copyright notice and license information have been removed. It is essential to ensure that the appropriate license information is retained in the file for legal compliance and attribution.

2. **ProfileInfo Struct and Methods:**
   - The `ProfileInfo` struct and associated methods have been removed. If these were not being used, it is appropriate to remove them. However, it's crucial to verify that these functions are genuinely unused and not required for any functionality.

3. **NewProfileInfo Function:**
   - The `NewProfileInfo` function has been removed along with the initialization of `functions` and `tagFunctions` maps. Ensure that these maps are not needed elsewhere in the codebase before removing them.

4. **getFunction, getTagFunction, getLocation, getTagLocation Methods:**
   - These methods have been removed, which were responsible for handling profile information. Confirm that there are no dependencies on these methods before deletion.

#### Suggestions for Improvement:
1. **License Information:**
   - Revert the removal of license information and copyright notice to comply with licensing requirements.

2. **Code Cleanup:**
   - Verify that the removed code is genuinely unused by conducting a thorough code review or using static code analysis tools.

3. **Documentation:**
   - Consider updating the comments or documentation to reflect the changes made and provide clarity on the purpose of the file and its contents.

4. **Review Dependencies:**
   - Ensure that no other parts of the codebase rely on the removed functions or structures. If needed, update or refactor the code to eliminate dependencies.

5. **Optimization:**
   - If there are other unused components or redundant code in the fileservice package, consider removing them in this cleanup process to improve code maintainability.

By addressing the mentioned points and ensuring the removal of unused code is accurate, the pull request can contribute to a cleaner and more efficient codebase.

**Here are review comments for file pkg/fileservice/profile_span.go:**
### Pull Request Review:

#### Title:
The title of the pull request is clear and concise, indicating the purpose of the changes made.

#### Body:
The body of the pull request provides some context about the type of PR and the issue it fixes. It also mentions that the PR is for cleaning up unused code, which is a good practice for code maintenance.

#### Changes:
1. The entire file `profile_span.go` has been removed in this pull request. This indicates that the code related to `SpanProfiler` and `ProfileSpan` structs, along with associated methods, has been removed from the codebase.

#### Feedback and Suggestions for Improvement:
1. **Unused Code Removal:**
   - Removing unused code is generally a good practice as it helps in reducing code clutter and improves code maintainability. However, it's important to ensure that the code being removed is truly unused and not required by any other part of the system.
   - Before removing the code, it's recommended to perform a thorough code review to confirm that the functionality provided by the `SpanProfiler` and `ProfileSpan` structs is no longer needed in the system.
   - If there is a possibility of reusing or repurposing the code in the future, consider commenting out the code instead of deleting it completely.

2. **Documentation and Comments:**
   - It's essential to have clear and concise comments in the code to explain the purpose of each function, struct, and method. This helps other developers understand the codebase easily.
   - If the code is being removed permanently, it's a good practice to update any associated documentation or comments to reflect the changes made.

3. **Code Optimization:**
   - While removing unused code is beneficial, it's also important to ensure that the codebase remains efficient and optimized.
   - Consider refactoring or restructuring the codebase to improve its readability, maintainability, and performance.
   - Look for opportunities to optimize existing code or introduce new features that can enhance the overall functionality of the system.

4. **Version Control and History:**
   - Before permanently deleting code, consider using version control tools like Git to keep a record of the changes made.
   - By using branches or tags in Git, you can preserve the history of the codebase and easily revert back to previous versions if needed.

5. **Security Considerations:**
   - While the changes in this pull request seem focused on code cleanup, it's crucial to always consider security implications when modifying the codebase.
   - Ensure that the removal of code does not introduce any security vulnerabilities or weaken the overall security posture of the system.

In conclusion, while the removal of unused code is a positive step towards maintaining a clean codebase, it's important to conduct a thorough review of the changes to ensure that no critical functionality is being removed inadvertently. Additionally, consider incorporating the feedback provided above to enhance the quality and efficiency of the codebase.

**Here are review comments for file pkg/fileservice/profile_span_test.go:**
### Pull Request Review:

#### Title:
The title of the pull request is clear and concise, indicating the intention to remove unused code from the fileservice.

#### Body:
The body of the pull request provides relevant information about the type of PR and the issue it fixes. It also explains the purpose of the PR, which is to clean up unused code.

#### Changes:
The changes in the pull request involve the removal of a test function `TestSpanProfiler` from the file `profile_span_test.go` in the `fileservice` package. The removed code includes test setup, assertions, and cleanup related to profiling spans.

#### Feedback and Suggestions:
1. **Unused Code Removal:**
   - The removal of unused code is a good practice to keep the codebase clean and maintainable.
   - The removed test function `TestSpanProfiler` seems to be related to profiling spans, but since it is no longer used, removing it is appropriate.

2. **Optimization:**
   - While removing unused code is beneficial, it's essential to ensure that the code being removed is genuinely unused.
   - It's recommended to double-check if any other parts of the codebase rely on the functionality provided by the removed test function. If there is a possibility of future use, consider commenting out the code instead of deleting it outright.

3. **Security Concerns:**
   - The removed code does not seem to introduce any security vulnerabilities based on the provided diff.
   - However, always be cautious when removing code, as it may inadvertently impact security if not thoroughly reviewed.

4. **Documentation:**
   - Consider updating relevant documentation or comments to reflect the removal of the `TestSpanProfiler` function, ensuring that the codebase remains well-documented.

5. **Testing:**
   - After removing the test function, ensure that other tests in the `fileservice` package or related areas are not affected by the removal.
   - Running the test suite after the removal can help validate that the changes do not introduce unexpected issues.

6. **Version Control:**
   - It's advisable to commit changes in smaller, logical chunks to make it easier to review and revert if necessary.
   - If there are multiple files or sections of code being removed, consider splitting them into separate commits for better traceability.

In conclusion, the pull request to remove unused code is a positive step towards codebase maintenance. By following the suggestions provided above, you can ensure that the changes are optimized and do not inadvertently impact other parts of the system.

**Here are review comments for file pkg/fileservice/profile_test.go:**
### Pull Request Review:

#### Title:
The title of the pull request is clear and concise, indicating the intention to remove unused code from the fileservice package.

#### Body:
The body of the pull request provides additional context by specifying the type of PR (Improvement), the related issue (#14908), and the purpose of the PR, which is to clean up unused code.

#### Changes:
The changes in the pull request involve the removal of a significant portion of code from the `profile_test.go` file in the `fileservice` package. The removed code includes test functions, benchmarks, and associated imports and comments.

#### Feedback and Suggestions:
1. **Unused Code Removal:**
   - The removal of unused code is a good practice to maintain a clean and efficient codebase. However, it's essential to ensure that the removed code is genuinely unused and does not have any potential impact on the functionality of the application.
   - Before removing code, it's recommended to perform a thorough review to confirm that the code is not needed for any existing or future functionality.

2. **License Information:**
   - The removed license information at the beginning of the file should be retained unless there is a specific reason for its removal. License information is crucial for maintaining legal compliance and providing clarity on the permitted use of the code.

3. **Test Functions and Benchmarks:**
   - The removed test functions and benchmarks may have been valuable for ensuring the correctness and performance of the fileservice package. It's important to verify whether these tests are redundant or if they need to be refactored or relocated to a different file.

4. **Documentation:**
   - While the focus of this PR is on code cleanup, it's always beneficial to update any relevant documentation to reflect the changes made. This ensures that other developers are aware of the modifications and their implications.

5. **Optimization:**
   - To optimize the changes made in this PR, consider organizing the remaining code in the `profile_test.go` file for better readability and maintainability. Group related test functions and benchmarks, add comments where necessary, and ensure consistent formatting.

6. **Review Process:**
   - It's advisable to involve team members or reviewers in the process of removing unused code to gather different perspectives and ensure that no critical functionality is inadvertently removed.

7. **Security Considerations:**
   - While not directly related to this PR, it's essential to conduct regular security reviews of the codebase to identify and address any potential security vulnerabilities.

In conclusion, while the removal of unused code is a positive step towards codebase maintenance, it's crucial to approach it with caution and thoroughness to prevent any unintended consequences. Additionally, maintaining a balance between code cleanup and preserving essential functionality is key to a healthy and efficient codebase.

**Review by zhangxu19830126 — APPROVED** [review]

**Review by triump2020 — APPROVED** [review]

**Pull request merged**
